tests/test_pin_lock_advanced.py

```diff
@@ -0,0 +1,220 @@
+            for name, config in self.mock_config["chipflow"]["silicon"]["power"].items():
+                power[name] = PadConfig(
+                    type=config["type"],
+                    loc=config["loc"]
+                )
+
+            # Create a Silicon config object
+            silicon_config = SiliconConfig(
+                process="ihp_sg13g2",
+                package="pga144",
+                pads=pads,
+                power=power
+            )
+
+            # Create the Chipflow config object with proper StepsConfig
+            from chipflow_lib.config_models import StepsConfig
+
+            steps_config = StepsConfig(
+                silicon="chipflow_lib.steps.silicon:SiliconStep"
+            )
+
+            chipflow_config = ChipFlowConfig(
+                project_name="test_project",
+                silicon=silicon_config,
+                steps=steps_config
+            )
+
+            # Create the full Config object
+            config_model = Config(chipflow=chipflow_config)
+
+            # Set up the mock model_validate to return our real Pydantic config object
+            mock_model_validate.return_value = config_model
+
+            # Set up parse_config to return the dict version
+            mock_parse_config.return_value = self.mock_config
+
+            # Mock top interfaces to return something simple
+            mock_interface = {
+                "component1": {
+                    "interface": {
+                        "members": {
+                            "uart": {
+                                "type": "interface",
+                                "members": {
+                                    "tx": {"type": "port", "width": 1, "dir": "o"},
+                                    "rx": {"type": "port", "width": 1, "dir": "i"}
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            mock_top_interfaces.return_value = ({}, mock_interface)
+
+            # Call lock_pins with mocked file operations
+            with mock.patch('builtins.print'), \
+                 mock.patch('pathlib.Path.exists', return_value=False), \
+                 mock.patch('chipflow_lib.pin_lock.LockFile.model_validate_json'):
+                lock_pins()
+
+            # Check that a lockfile was created
+            lockfile_path = Path('pins.lock')
+            self.assertTrue(lockfile_path.exists())
+
+            # Read the lockfile
+            with open(lockfile_path, 'r') as f:
+                lock_data = json.load(f)
+
+            # Verify it has the expected structure (Pydantic model)
+            self.assertIn("process", lock_data)
+            self.assertIn("package", lock_data)
+            self.assertIn("port_map", lock_data)
+            self.assertIn("metadata", lock_data)
+
+            # Verify process is correct
+            self.assertEqual(lock_data["process"], "ihp_sg13g2")
```

Comment:
this seems like it makes the  `assertIn` above redundant, if `process` isn't in `lock_data` then this will fail anyway

Reply:
yeah, though the fail would be less obvious.

Reply:
I prefer to leave the check in for neatness.